common/test/acceptance/tests/lms/test_account_settings.py

```diff
@@ -0,0 +1,482 @@
+        user_id = auto_auth_page.get_user_id()
+        return username, user_id
+
+    def assert_event_emitted_num_times(self, user_id, setting, num_times):
+        """
+        Verify a particular user settings change event was emitted a certain
+        number of times.
+        """
+        # pylint disable=no-member
+        super(AccountSettingsTestMixin, self).assert_event_emitted_num_times(
+            self.USER_SETTINGS_CHANGED_EVENT_NAME, self.start_time, user_id, num_times, setting=setting
+        )
+
+    def verify_settings_changed_events(self, username, user_id, events, table=None):
+        """
+        Verify a particular set of account settings change events were fired.
+        """
+        expected_referers = [self.ACCOUNT_SETTINGS_REFERER] * len(events)
+        for event in events:
+            event[u"user_id"] = long(user_id)
+            event[u"table"] = u"auth_userprofile" if table is None else table
+            event[u"truncated"] = []
+
+        self.verify_events_of_type(
+            username, self.USER_SETTINGS_CHANGED_EVENT_NAME, events,
+            expected_referers=expected_referers
+        )
+
+
+class DashboardMenuTest(AccountSettingsTestMixin, WebAppTest):
+    """
+    Tests that the dashboard menu works correctly with the account settings page.
+    """
+    def test_link_on_dashboard_works(self):
+        """
+        Scenario: Verify that the "Account Settings" link works from the dashboard.
+
+
+        Given that I am a registered user
+        And I visit my dashboard
+        And I click on "Account Settings" in the top drop down
+        Then I should see my account settings page
+        """
+        self.log_in_as_unique_user()
+        dashboard_page = DashboardPage(self.browser)
+        dashboard_page.visit()
+        dashboard_page.click_username_dropdown()
+        self.assertIn('Account Settings', dashboard_page.username_dropdown_link_text)
+        dashboard_page.click_account_settings_link()
+
+
+class AccountSettingsPageTest(AccountSettingsTestMixin, WebAppTest):
+    """
+    Tests that verify behaviour of the Account Settings page.
+    """
+    SUCCESS_MESSAGE = 'Your changes have been saved.'
+
+    def setUp(self):
+        """
+        Initialize account and pages.
+        """
+        super(AccountSettingsPageTest, self).setUp()
+        self.username, self.user_id = self.log_in_as_unique_user()
+        self.visit_account_settings_page()
+
+    def visit_account_settings_page(self):
+        """
+        Visit the account settings page for the current user.
+        """
+        self.account_settings_page = AccountSettingsPage(self.browser)
+        self.account_settings_page.visit()
+        self.account_settings_page.wait_for_ajax()
+
+    def test_page_view_event(self):
+        """
+        Scenario: An event should be recorded when the "Account Settings"
+           page is viewed.
+
```

Comment:
We have an odd mix of single and double quotes in this file. It seems that double quotes are more prevalent at least here.